src/shared/Microsoft.Git.CredentialManager/ApplicationBase.cs

```diff
@@ -86,18 +86,25 @@ public Task<int> RunAsync(string[] args)
 
         public static string GetEntryApplicationPath()
         {
-            // Assembly::Location always returns an empty string if the application was published as a single file
+            // Single file publishing does not exist with .NET Framework so
+            // we can just use reflection to get the entry assembly path.
+#if NETFRAMEWORK
+            return Assembly.GetEntryAssembly().Location;
+#endif
```

Comment:
Will this cause "unreachable code" errors in .NET Framework? Should we use an `#else`/`#endif` block to contain the rest of the code?